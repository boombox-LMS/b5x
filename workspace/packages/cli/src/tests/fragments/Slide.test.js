const {
  exampleSlideMarkup,
} = require("../../../dist/parser/componentBuilders/fragments/Slide");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Slide fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleSlideMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed slide fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed slide fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
