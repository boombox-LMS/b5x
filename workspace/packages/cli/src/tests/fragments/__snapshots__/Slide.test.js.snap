// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Slide fragment builds as expected The parsed slide fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Slide",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "15b560681e8c9367ef5211e8e28b4a10",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
